Ginger/Ginger/SolutionWindows/TargetApplicationsPage.xaml.cs

```diff
@@ -120,34 +125,44 @@ private void SaveHandler(object sender, RoutedEventArgs e)
 
         private void AddApplication(object sender, RoutedEventArgs e)
         {
-            AddApplicationPage AAP = new AddApplicationPage( WorkSpace.Instance.Solution);
+            AddApplicationPage AAP = new AddApplicationPage( WorkSpace.Instance.Solution);            
             AAP.ShowAsWindow();
         }
 
         private void ApplicationGrid_PreparingCellForEdit(object sender, DataGridPreparingCellForEditEventArgs e)
-        {
+        {        
             if (e.Column.DisplayIndex == 0)//App Name Column
             {
                 ApplicationPlatform currentApp = (ApplicationPlatform)xTargetApplicationsGrid.CurrentItem;
-                currentApp.NameBeforeEdit = currentApp.AppName;
+                currentApp.NameBeforeEdit = currentApp.AppName;                
             }
         }
-
+        private ObservableList<ApplicationPlatform> Applications;
         private void ApplicationGrid_CellEditEnding(object sender, DataGridCellEditEndingEventArgs e)
-        {
+        {           
             //Validate the name of the App is unique
-            if (e.Column.DisplayIndex == 0)//App Name Column
-            {
-                ApplicationPlatform currentApp = (ApplicationPlatform)xTargetApplicationsGrid.CurrentItem;
+            if (e.Column.DisplayIndex ==1)//App Name Column
+            {                
+                ApplicationPlatform currentApp = (ApplicationPlatform)xTargetApplicationsGrid.CurrentItem;                               
                 mSolution.SetUniqueApplicationName(currentApp);
 
                 if (currentApp.AppName != currentApp.NameBeforeEdit)
                 {
+                    if (currentApp.AppName == null || string.IsNullOrEmpty(currentApp.AppName.ToString()) || string.IsNullOrWhiteSpace(currentApp.AppName.ToString()))
```

Comment:
String is null or empty will do the null check, no need to explicitly check for null

Reply:
Removed extra condition and time complexity issue is handled.
Hence the comments stated are resolved